internal/gensupport/resumable.go

```diff
@@ -177,6 +177,22 @@ func (rx *ResumableUpload) Upload(ctx context.Context) (resp *http.Response, err
 				return prepareReturn(resp, err)
 			}
 
+			// Check for context cancellation or timeout once more. If more than one
+			// case in the select statement above was satisfied at the same time, Go
+			// will choose one arbitrarily.
+			// That can cause an operation to go through even if the context was
+			// canceled before or the timeout was reached.
+			select {
+			case <-ctx.Done():
+				if err == nil {
+					err = ctx.Err()
+				}
+				return prepareReturn(resp, err)
+			case <-quitAfter:
+				return prepareReturn(resp, err)
```

Comment:
In this branch should we also check ctx.Err you thing? Or will err already be non-nil here?

Reply:
It definitely should be non-nil; I don't think there is any way we could hit the per-chunk deadline (32s) without having at least one failed response first.